File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/UserQuitPacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.components.QuitState;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserQuitPacket implements ServerPacket {
  private int userId;
  private QuitState state;

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    writer.writeInt32(stream, userId);
    writer.writeUint8(stream, state.getValue());
  }

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_USER_QUIT;
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/UserQuitPacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.components.QuitState
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.core.ServerPacket
import pe.nanamochi.banchus.io.DataWriter

class UserQuitPacket(var userId: Int = 0, var state: QuitState = QuitState.GONE) : ServerPacket() {
    override val type = PacketType.BANCHO_USER_QUIT

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeInt32(out, userId)
        writer.writeUint8(out, state.value.toUByte())
    }
}
